src/features/class-graphql.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * GraphQL class file
+ *
+ * @package wp-curate
+ */
+
+namespace Alley\WP\WP_Curate\Features;
+
+use Alley\WP\Types\Feature;
+use WPGraphQL\AppContext;
+use GraphQL\Type\Definition\ResolveInfo;
+use WPGraphQL\Data\Connection\PostObjectConnectionResolver;
+
+/**
+ * Add support for GraphQL, if WPGraphQL plugin exists.
+ */
+final class GraphQL implements Feature {
+	/**
+	 * Post types allowed in GraphQL.
+	 */
+	private $allowed_post_types;
+
+	/**
+	 * Get the GraphQL Type by Post Type.
+	 *
+	 * @param  string $post_type_string Name of the post type. Ex: 'post'.
+	 *
+	 * @return string GraphQL Type name. Ex: 'Post'.
+	 */
+	private function get_graphql_type_by_post_type( $post_type_string ): string {
+		$post_type_object = get_post_type_object( $post_type_string );
+
+		if ( empty( $post_type_object ) ) {
+			return '';
+		}
+
+		// Only return a GraphQL type for allowed post types in WP Curate.
+		if ( ! in_array( $post_type_string, $this->allowed_post_types, true ) ) {
```

Comment:
This check should come first, because it doesn't rely on fetching data from somewhere else (e.g., get_post_type_object).

Reply:
Updated in d6644c4e6727953425c9ad0899f4de0f041f1806